bionty/models.py

```diff
@@ -438,25 +435,29 @@ def public(
                 source_name = source.name
                 version = source.version
 
+        if source is not None and source.dataframe_artifact_id is not None:
+            # if source is a DataFrame artifact, return the static reference
+            return StaticReference(source)
+        if source is None:
+            kwargs = {
+                "entity": cls.__get_name_with_module__(),
+                "currently_used": True,
+            }
+            if organism is not None:
+                if isinstance(organism, DBRecord):
+                    kwargs["organism"] = organism.name
+                else:
+                    kwargs["organism"] = organism
+            source = Source.filter(**kwargs).first()
+            if source and source.dataframe_artifact_id is not None:
```

Comment:
The "dataframe_artifact" as an attribute name sounds strange to me, but if there really isn't a better name, so be it.

Reply:
It's a foreign key only for internal use.